plugins/outputs/splunk/README.md

```diff
@@ -0,0 +1,81 @@
+# Splunk Output Plugin
+
+This plugin writes metrics to a [Splunk Metric Index](https://docs.splunk.com/Documentation/Splunk/latest/Indexer/Setupmultipleindexes#Create_metrics_indexes) via a [Splunk HTTP Event Collector (HEC)](http://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector).  This requires an `Authorization Token` which will be created when you setup your Splunk HEC Token.
+
+Notes:
+1. If any point value cannot be converted to a float64, that metric will be skipped.
+2. Metrics are grouped by converting any `_` characters to `.` in the Metric Name.
+
+
+
+### Configuration:
+
+```toml
+# Configuration for Splunk server to send metrics to
+[[outputs.splunk]]
+  ## REQUIRED
+  ## URL of the Splunk Enterprise HEC endpoint (i.e.: http://localhost:8088/services/collector)
+  SplunkUrl = "http://localhost:8088/services/collector"
+
+  ## REQUIRED
+  ## Splunk Authorization Token for sending data to a Splunk HTTPEventCollector (HEC).
+  ##   Note:  This Token should map to a 'metrics' index in Splunk.  
+  AuthString = "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
+
+  ## OPTIONAL:  prefix for metrics keys
+  #prefix = "my.specific.prefix."
+
+  ## OPTIONAL:  whether to use "value" for name of simple fields
+  ##  Default is false which will result in using only the measurement name as the metric name, not "value"
+  #simple_fields = false
+
+  ## OPTIONAL:  character to use between metric and field name.  defaults to . (dot)
+  #metric_separator = "."
+
+  ## OPTIONAL:  Convert metric name paths to use metricSeperator character
+  ## When true will convert all _ (underscore) chartacters in final metric name
+  #convert_paths = false
+
+  ## OPTIONAL:  Replace special characters in metric names with "-".  
+  ## This can be useful if metric names contain special characters  
+  #replace_special_chars = false
+
+  ## OPTIONAL:  Use Regex to sanitize metric and tag names from invalid characters
+  ## Regex is more thorough, but significantly slower
+  #use_regex = false
```

Comment:
`replace_special_chars` and `use_regex` combined cover all the possible cases, but maybe a single parameter would be better.
E.G.
```
name_sanatize = "none"
name_sanatize = "regex"
name_sanatize = "transliterate"
```
Note, this is not a requirement, or even a suggestion on the name, just throwing out ideas. @danielnelson might have a stronger preference.

Reply:
by keeping them separate the user will have the option to use both regex and the predefined character set.  I'm going to leave this one as is.